eclair-core/src/main/scala/fr/acinq/eclair/channel/InteractiveTxBuilder.scala

```diff
@@ -387,7 +387,11 @@ private class InteractiveTxBuilder(replyTo: ActorRef[InteractiveTxBuilder.Respon
           log.info("added {} inputs and {} outputs to interactive tx", inputDetails.usableInputs.length, outputs.length)
           // We unlock the unusable inputs from previous iterations (if any) as they can be used outside of this session.
           unlock(unusableInputs.map(_.outpoint))
-          stash.unstashAll(buildTx(FundingContributions(inputDetails.usableInputs, outputs)))
+          // The initiator's serial IDs must use even values and the non-initiator odd values.
+          val serialIdParity = if (fundingParams.isInitiator) 0 else 1
+          val txAddInputs = inputDetails.usableInputs.zipWithIndex.map { case (input, i) => input.copy(serialId = UInt64(2 * i + serialIdParity)) }
+          val txAddOutputs = outputs.zipWithIndex.map { case (output, i) => output.copy(serialId = UInt64(2 * (i + txAddInputs.length) + serialIdParity)) }
```

Comment:
For fun... yours is safer!

```suggestion
          val iterator = (serialIdParity to Int.MaxValue by 2).iterator
          val txAddInputs = inputDetails.usableInputs.zip(iterator).map { case (input, i) => input.copy(serialId = UInt64(i)) }
          val txAddOutputs = outputs.zip(iterator).map { case (output, i) => output.copy(serialId = UInt64(i)) }
```